include/clang/AST/ASTImporter.h

```diff
@@ -132,6 +133,9 @@ class TypeSourceInfo;
     /// The file managers we're importing to and from.
     FileManager &ToFileManager, &FromFileManager;
 
+    /// The ASTUnit that this importer belongs to, if any.
+    ASTUnit *Unit;
```

Comment:
Unit->getASTContext() should return a ref which is equivalent to `ToContext`, right?
Then perhaps we should assert on this in the ctor.
Plus, I think the comment should indicate that this is associated with the "to" context.

Reply:
This is the `FromContext`, not the To. The ASTImporter's in `CrossTranslationUnitContext` have all the same `ToContext` (and same To AST) and a different `FromContext` (that belongs to the loaded From AST). The `ASTUnit` objects manage the loaded From AST's.

Reply:
Okay, then we should assert that Unit->getASTContext() should return a ref which is equivalent to FromContext

Reply:
I do not like to put assert into `ASTImporter` for this, the passed `ASTUnit` pointer is just an "arbitrary value" here that is put into ImportedFileIDs (working of the ASTImporter itself does not depend on any value of it). (And only for this assert the ASTUnit header must be used.) The assert can be put into where the ASTImporter is created.